tensorrt_llm/_torch/modules/attention.py

```diff
@@ -544,6 +563,10 @@ def forward(
             q_ctx = q[:num_ctx_tokens, ...]
             compressed_kv_ctx = compressed_kv[:num_ctx_tokens, ...]
             k_pe_ctx = k_pe[:num_ctx_tokens, ...]
+            if self.rotary_emb is not None and position_ids is not None:
```

Comment:
```suggestion
            if apply_rotary_emb:
```
Add a bool for `apply_rotary_emb = not self.enable_rope_fusion` and a check before line 562
```
if apply_rotary_emb:
    assert self.rotary_emb is not None and position_ids is not None
```

Reply:
Updated. Since we always create `self.rotary_emb` when `apply_rotary_emb` is True, I only add `assert position_ids is not None`.